File: microservers/immobilierservice/src/main/java/io/multi/immobilierservice/dto/ProfilImmoRequest.java
package io.multi.immobilierservice.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Data
public class ProfilImmoRequest {

    @NotBlank
    @Pattern(regexp = "PROPRIETAIRE_SIMPLE|DEMARCHEUR|AGENT_AGENCE",
            message = "typeProfil doit être PROPRIETAIRE_SIMPLE, DEMARCHEUR ou AGENT_AGENCE")
    private String typeProfil;

    private String agenceUuid;             // requis si AGENT_AGENCE

    @Size(max = 500)
    private String documentsKycUrl;        // URL externe pour l'instant (MinIO en Phase 5)

    @Size(max = 2000)
    private String bio;

    @Size(max = 20)
    private String telephoneContact;
}

File: mobile_app/lib/features/immo/models/profil_immo_request.dart
/// DTO de requête pour `POST /immo/profils` et `PUT /immo/profils/{uuid}`.
///
/// Conforme à `ProfilImmoRequest.java` côté backend :
/// - typeProfil : PROPRIETAIRE_SIMPLE | DEMARCHEUR | AGENT_AGENCE (NotBlank)
/// - agenceUuid : requis SI typeProfil=AGENT_AGENCE (non utilisé en 15.2e-1,
///   AGENT_AGENCE hors scope MVP)
/// - bio : ≤2000 chars
/// - telephoneContact : ≤20 chars
class ProfilImmoRequest {
  final String typeProfil;
  final String? agenceUuid;
  final String? bio;
  final String? telephoneContact;

  const ProfilImmoRequest({
    required this.typeProfil,
    this.agenceUuid,
    this.bio,
    this.telephoneContact,
  });

  Map<String, dynamic> toJson() => {
        'typeProfil': typeProfil,
        if (agenceUuid != null) 'agenceUuid': agenceUuid,
        if (bio != null && bio!.isNotEmpty) 'bio': bio,
        if (telephoneContact != null && telephoneContact!.isNotEmpty)
          'telephoneContact': telephoneContact,
      };
}
